src/main/java/se/sundsvall/messaging/api/model/request/EmailRequest.java

```diff
@@ -21,77 +23,89 @@
-        String emailAddress,
-
-        @NotBlank
-        @Schema(description = "E-mail subject", requiredMode = REQUIRED)
-        String subject,
-
-        @Schema(description = "E-mail plain-text body")
-        String message,
-
-        @ValidBase64(nullable = true)
-        @Schema(description = "E-mail HTML body (BASE64-encoded)")
-        String htmlMessage,
-
-        @Valid
-        @Schema(description = "Sender")
-        Sender sender,
-
-        @ArraySchema(schema = @Schema(implementation = Attachment.class))
-        List<@Valid Attachment> attachments) {
-
-    @With
-    @Builder(setterPrefix = "with")
-    @Schema(name = "EmailRequestParty")
-    public record Party(
-
-        @ValidUuid(nullable = true)
-        @Schema(description = "The message party id", format = "uuid")
-        String partyId,
-
-        @Schema(description = "External references")
-        List<@Valid ExternalReference> externalReferences) {
-    }
-
-    @With
-    @Builder(setterPrefix = "with")
-    @Schema(name = "EmailSender", description = "Attachment")
-    public record Sender(
-
-        @NotBlank
-        @Schema(description = "The sender of the e-mail")
-        String name,
-
-        @Email
-        @NotBlank
-        @Schema(description = "Sender e-mail address", example = "sender@sender.se")
-        String address,
-
-        @Email
-        @Schema(description = "Reply-to e-mail address", example = "sender@sender.se")
-        String replyTo) { }
-
-    @With
-    @Builder(setterPrefix = "with")
-    @Schema(name = "EmailAttachment", description = "Attachment")
-    public record Attachment(
-
-        @NotBlank
-        @Schema(description = "The attachment filename", example = "test.txt", requiredMode = REQUIRED)
-        String name,
-
-        @Schema(description = "The attachment content type", example = "text/plain")
-        String contentType,
-
-        @ValidBase64
-        @Schema(description = "The attachment (file) content as a BASE64-encoded string", example = "aGVsbG8gd29ybGQK", requiredMode = REQUIRED)
-        String content) { }
+	@Valid
+	@Schema(description = "Party")
+	Party party,
+
+	@Email
+	@NotBlank
+	@Schema(description = "Recipient e-mail address", requiredMode = REQUIRED)
+	String emailAddress,
+
+	@NotBlank
+	@Schema(description = "E-mail subject", requiredMode = REQUIRED)
+	String subject,
+
+	@Schema(description = "E-mail plain-text body")
+	String message,
+
+	@ValidBase64(nullable = true)
+	@Schema(description = "E-mail HTML body (BASE64-encoded)")
+	String htmlMessage,
+
+	@Valid
+	@Schema(description = "Sender")
+	Sender sender,
+
+	@ArraySchema(schema = @Schema(implementation = Attachment.class))
+	List<@Valid Attachment> attachments,
+
+	@Schema(description = "Headers")
+	Map<Header, List<@Pattern(regexp = "^<.{1,1000}@.{1,1000}>$", message = "Message-ID need to begin with '<' contain '@' and end with '>'") String>> headers) {
```

Comment:
Just curious, won't this validation trigger and reject payload if client chooses to send other headers than message-id? I.e. if client chooses to send reply_to, wont this validation reject that header?

If the pattern is valid for all header values, maybe change the error message to a more "general" one, for example `header values must begin with '<', contain '@' and end with '>'`?

Reply:
The three headers, In-Reply-To, References and Message-ID should all contain either one or more 'Message-ID's.